app/utils/store.py

```diff
@@ -31,17 +37,29 @@ def put(self, service_id, document_stream, *, mimetype, verification_email=None)
 
         extra_kwargs = {}
         if verification_email:
+            # Make super sure that the email is in our standardised format
+            try:
+                verification_email = clean_and_validate_email_address(verification_email)
+            except InvalidEmailError as e:
+                raise ValueError(str(e))
+
             hashed_recipient_email = self._hasher.hash(verification_email)
             extra_kwargs['Metadata'] = {"hashed-recipient-email": hashed_recipient_email}
 
+        if retention_period:
+            # Make super sure that our retention period is in our standardised format
+            retention_period = clean_and_validate_retention_period(retention_period)
```

Comment:
Same with double-cleaning and validating retention period. I don't think I understand why we do this twice. Am I missing something?

Reply:
It's because this function _could_ get called from somewhere else, and we want to be absolutely sure it's in the correct format.